SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityOfferHolder.java

```diff
@@ -25,25 +26,31 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(SingularityMesosScheduler.class);
 
-  private final Protos.Offer offer;
+  private final List<Protos.Offer> offers;
   private final List<SingularityTask> acceptedTasks;
   private final Set<SingularityPendingTaskId> rejectedPendingTaskIds;
   private List<Resource> currentResources;
+  private Set<String> roles;
 
   private final String rackId;
+  private final String slaveId;
+  private final String hostname;
   private final String sanitizedHost;
   private final String sanitizedRackId;
 
   private final Map<String, String> textAttributes;
   private final Map<String, String> reservedSlaveAttributes;
 
-  public SingularityOfferHolder(Protos.Offer offer, int taskSizeHint, String rackId, Map<String, String> textAttributes, Map<String, String> reservedSlaveAttributes) {
+  public SingularityOfferHolder(List<Protos.Offer> offers, int taskSizeHint, String rackId, String slaveId, String hostname, Map<String, String> textAttributes, Map<String, String> reservedSlaveAttributes) {
     this.rackId = rackId;
-    this.offer = offer;
-    this.acceptedTasks = Lists.newArrayListWithCapacity(taskSizeHint);
-    this.currentResources = offer.getResourcesList();
+    this.slaveId = slaveId;
+    this.hostname = hostname;
+    this.offers = offers;
+    this.roles = MesosUtils.getRoles(offers.get(0));
+    this.acceptedTasks = Lists.newArrayListWithExpectedSize(taskSizeHint);
+    this.currentResources = offers.size()  > 1 ? MesosUtils.combineResources(offers.stream().map(Protos.Offer::getResourcesList).collect(Collectors.toList())) : offers.get(0).getResourcesList();
```

Comment:
Wonder if it's possible to have two currentResources for this? One for the individual offers (as originally given by mesos) and one for the collection of same host offers? The check could then try all the single offers and check the combined offers if no single offers were good enough. 

It could help launch more tasks if the smaller offers were used instead of declined. The tough part would be that the combinedCurrentResources would need to be calculated each time before it's checked since some single offers could be utilized for a task in a later iteration.